### PR TITLE
fix: lambda role problem for sagemaker

### PR DESCRIPTION
**Pull request by alvindaiyan**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: infrastructure/src/sd-inference/sagemaker-endpoints-create.ts [hunk 1]
```diff
@@ -153,12 +155,11 @@ export class CreateSagemakerEndpointsApi {
...
 
-    const newRole = new Role(this.scope, `${this.baseId}-role`, {
-      assumedBy: new CompositePrincipal(
-        new ServicePrincipal('sagemaker.amazonaws.com'),
-        new ServicePrincipal('lambda.amazonaws.com'),
-      ),
-    });
+    const lambdaStartDeployRole = <iam.Role>iam.Role.fromRoleName(
+      this.scope,
+      'createSagemakerEpRole',
```
Comment (yike5460) [inline infrastructure/src/sd-inference/sagemaker-endpoints-create.ts:160]: I assume this is the core modification that we create a RETAIN role with fix name

**Review by yike5460 — COMMENTED** [review]
lgtm, just one case need to be clarified.

File: infrastructure/src/sd-inference/sagemaker-endpoints-create.ts (hunk 1 above)
Comment (alvindaiyan) [reply infrastructure/src/sd-inference/sagemaker-endpoints-create.ts:160]: yes. the role is created in the shared stack

File: aws_extension/sagemaker_ui.py [hunk 2]
```diff
@@ -798,6 +798,9 @@ def process_result_by_inference_id(inference_id):
         return image_list, info_text, plaintext_to_html(infotexts), infotexts
     else:
         logger.debug(f"get_inference_job resp is {resp}")
+        if 'taskType' not in resp:
```
[the remark this answers is not in the document]
Comment (alvindaiyan) [reply aws_extension/sagemaker_ui.py:801]: sometimes, I discover the very first request to newly created endpoint would fail for 'endpoint timeout ' error

File: scripts/main.py [hunk 3]
```diff
@@ -226,13 +224,75 @@ def _check_cn(model_state, model_selected, k):
                                self.controlnet_components[f'{cn_txt2img_or_img2img}_controlnet_dropdown_batch'][i][0]
                            ])
                 cn_radio_tuple[1] = True
+
+        # xyz components:
+        if type(component) == ToolButton \
```
Comment (yike5460) [inline scripts/main.py:229]: this part should be a another PR to fix xyz issue

**Review by yike5460 — COMMENTED** [review]
make sure test case passed and raise separate PR for xyz feature
